trlx/trainer/nn/ppo_models.py

```diff
@@ -327,9 +340,12 @@ def __init__(
                 lm_head=self.base_model.lm_head,
             )
         # Cache `transformer.forward` args for general use (avoids incompatible args across architectures)
-        self.base_model_transformer_args = inspect.getfullargspec(
-            self.base_model.transformer.forward
-        ).args
+        if base_model_transformer_args is not None:
+            self.base_model_transformer_args = base_model_transformer_args
```

Comment:
@artek0chumak could you explain why this variable had to be introduced? Maybe as a more general way to control `__call__` method, we can get it done on the user-side by changing `base_model` instead

Reply:
That can be done because all transformers models have to use hf's interface. However, some of them(like Petals) cannot use all the passed inputs, like the attention masks.

I may remove this change and can resolve the issue on Petals' side. Will discuss this with my teammates.

Reply:
Any resolution?

Reply:
Yes, Petals will be changed, so I will remove these changes.